### PR TITLE
Update 5 fuzzy cases

### PR DESCRIPTION
**Pull request by lecranek**
HobList: copy source code of static functions from edk2 repo

ParseMmioExitInstructions: include source code c file from edk2 repo for
static functions

DxeCapsuleLibFmp: add QuickSort.c to BaseLibHost

TestTcg2MeasureGptTable&TestTcg2MeasurePeImage:
  previously cannot be run on Linux
root cause:
  in api gBS->InstallMultipleProtocolInterfaces,
  the method to parse variable paramters `Protocol = VA_ARG (Args, EFI_GUID *);`
  on edk2-staging/HBFA/UefiHostTestPkg/Library/UefiBootServicesTableLibHost/Handle.c:L560
  will cause segmentaion fault
solution:
  change to use api gBS->InstallProtocolInterface

Signed-off-by: Wang Chenhe <chenhe.wang@intel.com>

### CODE REVIEW

File: HBFA/UefiHostFuzzTestCasePkg/TestCase/MdeModulePkg/Library/SmmLockBoxLib/UpdateLockBoxTestCase/TestUpdateLockBoxFuzzLength.c [hunk 1]
```diff
@@ -79,7 +79,8 @@ RunTestHarness(
     0xdea652b0, 0xd588, 0x4c55, { 0xb5, 0xb4, 0xc6, 0x82, 0xe7, 0xa0,0x3d }
   };
 
-  SmmLockBoxSmmConstructor(NULL, NULL);
+  // commented due to no implementation
```
Comment (liuw1) [inline HBFA/UefiHostFuzzTestCasePkg/TestCase/MdeModulePkg/Library/SmmLockBoxLib/UpdateLockBoxTestCase/TestUpdateLockBoxFuzzLength.c]: Please do not modify the legacy test case

File: HBFA/UefiHostFuzzTestCasePkg/TestCase/MdeModulePkg/Library/SmmLockBoxLib/UpdateLockBoxTestCase/TestUpdateLockBoxFuzzOffset.c [hunk 2]
```diff
@@ -82,7 +82,8 @@ RunTestHarness(
     0xdea652b0, 0xd588, 0x4c55, { 0xb5, 0xb4, 0xc6, 0x82, 0xe7, 0xa0,0x3d }
   };
 
-  SmmLockBoxSmmConstructor(NULL, NULL);
+  // commented due to no implementation
```
Comment (liuw1) [inline HBFA/UefiHostFuzzTestCasePkg/TestCase/MdeModulePkg/Library/SmmLockBoxLib/UpdateLockBoxTestCase/TestUpdateLockBoxFuzzOffset.c]: Please do not modify the legacy test case

File: HBFA/UefiHostFuzzTestCasePkg/TestCase/OvmfPkg/Library/CcExitLib/TestParseMmioExitInstructions.c [hunk 3]
```diff
@@ -21,6 +21,7 @@
 #include <Library/MemoryAllocationLib.h>
 #include <Library/PcdLib.h>
 #include <Library/CcExitLib.h>
+#include <Library/CcExitLib/CcExitVeHandler.c>  // TODO: temporary hack for static functions
```
Comment (liuw1) [inline HBFA/UefiHostFuzzTestCasePkg/TestCase/OvmfPkg/Library/CcExitLib/TestParseMmioExitInstructions.c]: Remove todo

File: HBFA/UefiHostFuzzTestCasePkg/TestCase/OvmfPkg/Library/TdxStartupLib/TestHobList.c [hunk 4]
```diff
@@ -23,28 +23,233 @@
...
 #include <IndustryStandard/Tdx.h>
+#include <Pi/PrePiHob.h>
 
 #define GET_GPAW_INIT_STATE(INFO)  ((UINT8) ((INFO) & 0x3f))
 
 #define TOTAL_SIZE   (512 * 1024)
 #define BLOCK_SIZE   (512)
 #define IO_ALIGN     (1)
 
+// TODO: copied from SecTdxHelper.c, temporary hack for static functions
```
Comment (liuw1) [inline HBFA/UefiHostFuzzTestCasePkg/TestCase/OvmfPkg/Library/TdxStartupLib/TestHobList.c]: remove todo

File: HBFA/UefiHostFuzzTestCasePkg/UefiHostFuzzTestCasePkg.dsc [hunk 5]
```diff
@@ -219,7 +219,7 @@
    PlatformInitLib|OvmfPkg/Library/PlatformInitLib/PlatformInitLib.inf
    PeilessStartupLib|OvmfPkg/Library/PeilessStartupLib/PeilessStartupLib.inf
    RegisterFilterLib|MdePkg/Library/RegisterFilterLibNull/RegisterFilterLibNull.inf
-   UefiCpuLib|UefiCpuPkg/Library/BaseUefiCpuLib/BaseUefiCpuLib.inf
+   #UefiCpuLib|UefiCpuPkg/Library/BaseUefiCpuLib/BaseUefiCpuLib.inf
```
Comment (liuw1) [inline HBFA/UefiHostFuzzTestCasePkg/UefiHostFuzzTestCasePkg.dsc:222]: Please remove it

**Review by liuw1 — CHANGES_REQUESTED** [review]

File: HBFA/UefiHostFuzzTestCasePkg/UefiHostFuzzTestCasePkg.dsc (hunk 5 above)
Comment (lecranek) [reply HBFA/UefiHostFuzzTestCasePkg/UefiHostFuzzTestCasePkg.dsc:222]: Hi @liuw1, this library has been removed from edk2, do we need to keep this line uncommented?

File: HBFA/UefiHostFuzzTestCasePkg/TestCase/MdeModulePkg/Library/SmmLockBoxLib/UpdateLockBoxTestCase/TestUpdateLockBoxFuzzLength.c (hunk 1 above)
Comment (lecranek) [reply HBFA/UefiHostFuzzTestCasePkg/TestCase/MdeModulePkg/Library/SmmLockBoxLib/UpdateLockBoxTestCase/TestUpdateLockBoxFuzzLength.c]: fixed in [0148b9e](https://github.com/tianocore/edk2-staging/pull/386/commits/0148b9e1fc1d0947610f4bb1da22a6b373d9258e)

File: HBFA/UefiHostFuzzTestCasePkg/TestCase/MdeModulePkg/Library/SmmLockBoxLib/UpdateLockBoxTestCase/TestUpdateLockBoxFuzzOffset.c (hunk 2 above)
Comment (lecranek) [reply HBFA/UefiHostFuzzTestCasePkg/TestCase/MdeModulePkg/Library/SmmLockBoxLib/UpdateLockBoxTestCase/TestUpdateLockBoxFuzzOffset.c]: fixed in [0148b9e](https://github.com/tianocore/edk2-staging/pull/386/commits/0148b9e1fc1d0947610f4bb1da22a6b373d9258e)

File: HBFA/UefiHostFuzzTestCasePkg/TestCase/OvmfPkg/Library/CcExitLib/TestParseMmioExitInstructions.c (hunk 3 above)
Comment (lecranek) [reply HBFA/UefiHostFuzzTestCasePkg/TestCase/OvmfPkg/Library/CcExitLib/TestParseMmioExitInstructions.c]: fixed in [0148b9e](https://github.com/tianocore/edk2-staging/pull/386/commits/0148b9e1fc1d0947610f4bb1da22a6b373d9258e)

File: HBFA/UefiHostFuzzTestCasePkg/TestCase/OvmfPkg/Library/TdxStartupLib/TestHobList.c (hunk 4 above)
Comment (lecranek) [reply HBFA/UefiHostFuzzTestCasePkg/TestCase/OvmfPkg/Library/TdxStartupLib/TestHobList.c]: fixed in [0148b9e](https://github.com/tianocore/edk2-staging/pull/386/commits/0148b9e1fc1d0947610f4bb1da22a6b373d9258e)

**Review by liuw1 — APPROVED** [review]

**Comment by liuw1** [comment]
LGTM

**Pull request merged**